### DISCUSSION TITLE
Help please: How to create a standalone help file using SHFB

### DISCUSSION THREAD

**Discussion by BriseEdu** [EWSoftware/SHFB#995] [Q&A] [answered]
Hello friends,

I am trying to create help files with SHFB for my VS2022 projects. I downloaded the latest SHFL version and installed it successfully.

I then created the help file in Website (HTML/ASP.Net) format. However, I don't know how to make it portable like a standalone file so that I can easily share it with other people without referring to a website. Any ideas? 

(I tried to open index.html under Help/ directory generated by SHFB, hoping it would be the root or starting page for all other pages. However, it was not. It was not an index for all classes from all folders. Rather, it only displays the classes for certain folders/data model)

I then tried to see if I could create a standalone help file, like HTML Help 1 and Open XML formats. However, I could not find  those formats in the "Build these help file formats:" box for me to build, please see the attached picture. And I could not find a way to add those formats into the box. Any ideas?

Thank you very much in advance.

Andrew

![image](https://github.com/EWSoftware/SHFB/assets/117489376/5d550560-fa97-475e-8900-5850d1b4663e)

**Comment by EWSoftware** [comment, marked as answer]
Help formats are determined by the selected presentation style.  The only portable formats are Help 1 (CHM) and Open XML.  You'll need to use the VS2013 presentation style for the Help 1 format as that is the only one that supports it.  The Help 1 viewer is too old to support the newer HTML features so isn't supported by the Default 2022 presentation style.  Open XML is a unique format and has its own presentation style.  Use the Presentation Style property to switch between them.

**Reply by BriseEdu** [reply]
Thank you! I was able to create both Help 1 (CHM) and Open XML format files. That was great! BTW, when I opened them, Help 1 (CHM) worked well while Word (Open XML .docx) was very slow.

As for the Website format help files, what are the steps to deploy it to a local IIS server so that other people may access it if they need? (I tried to create a website pointing the Help\ directory created by SHFB, and it didn't work. The index.html seemes not a home page.)

**Reply by EWSoftware** [reply]
You'll need to create a site and deploy it to your web server however that is accomplished.  That isn't something I can comment on specifically since I'm not that familiar with IIS.  I assume it would be done using the tools provided with the web server.

**Reply by BriseEdu** [reply]
Thank you and I hope you had a nice weekend. As you know, all the Help files are generated under \Help directory. My problem was that I couldn't figure out which html/htm file is the homepage so that users can navigate to all other pages from it. (I can use Sandcastle Help File Builder GUI to open the homepage by clicking on Documentation\View Help File\View Help File or Documentation\View Help File\View Website (Local Dev Web Server) menu bar. But I could not find that homepage when looking at the files under \Help.)

**Reply by EWSoftware** [reply]
Index.html is always the default page.  It will redirect to the default topic automatically.
